### ISSUE TITLE
BigDecimal#whole?

### ISSUE THREAD

**Issue by dorianmariecom** [ruby/bigdecimal#291]
e.g. `round == self`

Would be nice for things like skiping the `.0` in `1.0`

**Comment by dorianmariecom** [comment]
And `#whole` would be alias of `#round` or `#floor` or `#ceil` or `#truncate`

**Comment by mrkn** [comment]
@dorianmariecom What specific changes are you proposing and how do you suggest we implement them? I'm having difficulty understanding from your current explanation, so could you please provide more detailed clarification?

**Comment by dorianmariecom** [comment]
e.g.

```
class BigDecimal
  def whole
    round
  end

  def whole?
    whole == self
  end
end
```

So a `whole` method to get the whole part of a number and `whole?` to see if that number is whole

**Comment by Justman100** [comment]
@mrkn

**Comment by tompng** [comment]
A method to get the integer part already exists.

Integer part / Integral part / Whole number part
```ruby
# doc: Return the integer part of the number, as a BigDecimal.
BigDecimal('12.34').fix #=> 0.12e2
BigDecimal('-12.34').fix #=> -0.12e2
```

Fractional part / Decimal part
```ruby
# doc: Return the fractional part of the number, as a BigDecimal.
BigDecimal('12.34').frac #=> 0.34e0
BigDecimal('-12.34').frac #=> -0.34e0
```

`.frac.zero?` can be used instead of the proposed `whole?` method.
```ruby
x = BigDecimal('12.34')
y = BigDecimal('123400')
x.frac.zero? #=> false
y.frac.zero? #=> true
```

**Comment by tompng** [comment]
I don't think `whole` and `whole?` is a good name.
`whole number part ≒ integer part = integral part` this is ok
`whole number = positive or non-negative integers` `.whole_number?` might be ok, but there is an ambiguity of zero and it can't be used as a integer check for negative numbers.
`whole = all, entire` of a BigDecimal does not make sense.

There is a confusing method `Numeric#integer?` that checks if the receiver is an instance of Integer class. Always returns false for BigDecimal.

**Comment by mrkn** [comment]
@dorianmariecom 
I agree with @tompng that `whole` isn't the right word for the name.

In addition, I'm not convinced that you really need the `whole?` method, because it simply discards the result of `round`.  You really don't need to keep the value, right? I suspect that you want to reuse the result of `round` if `whole?` returns true.  If so, you don't have to use `whole?` to avoid calling `round` twice on the same BigDecimal number.

Do you actually need the `whole?` method?
